src/spi/norflash.rs

```
@@ -0,0 +1,300 @@
+        start_transfer!(self, &mut nor_data);
+        Ok(())
+    }
+
+    fn nor_read_jedec_id(&mut self) -> Result<[u8; 3], Self::Error> {
+        let mut read_buf: [u8; 3] = [0, 0, 0];
+        let mut nor_data = SpiNorData {
+            mode: Jesd216Mode::Mode111,
+            opcode: 0x9F,
+            dummy_cycle: 0,
+            addr: 0,
+            addr_len: 0,
+            data_len: 0,
+            rx_buf: &mut read_buf,
+            tx_buf: &[],
+            data_direct: SPI_NOR_DATA_DIRECT_READ,
+        };
+        start_transfer!(self, &mut nor_data);
+        Ok([read_buf[0], read_buf[1], read_buf[2]])
+    }
+
+    fn nor_sector_erase(&mut self, address: u32) -> Result<(), Self::Error> {
+        self.nor_write_enable()?;
+        if self.nor_sector_aligned(address) == true {
+            let mut nor_data = SpiNorData {
+                mode: Jesd216Mode::Mode111,
+                opcode: norflash::SPI_NOR_CMD_SE,
+                dummy_cycle: 0,
+                addr: address,
+                addr_len: 3,
+                data_len: 0,
+                tx_buf: &[],
+                rx_buf: &mut [],
+                data_direct: SPI_NOR_DATA_DIRECT_WRITE,
+            };
+            start_transfer!(self, &mut nor_data);
+        }
+        Ok(())
+    }
+
+    fn nor_page_program(&mut self, address: u32, data: &[u8]) -> Result<(), Self::Error> {
+        self.nor_write_enable()?;
+        if self.nor_sector_aligned(address) == true {
+            let mut nor_data = SpiNorData {
+                mode: Jesd216Mode::Mode111,
+                opcode: norflash::SPI_NOR_CMD_PP,
+                dummy_cycle: 0,
+                addr: address,
+                addr_len: 3,
+                data_len: data.len() as u32,
+                tx_buf: data,
+                rx_buf: &mut [],
+                data_direct: SPI_NOR_DATA_DIRECT_WRITE,
+            };
+            start_transfer!(self, &mut nor_data);
+            self.nor_wait_until_ready();
```

Please handle errors here. Use cargo clipppy to identify other places where errors are not being handled.

---

will add error handling

---

Push the fixes back

---

added error handling